File: vue-spotify-app.Classes/TrackFilter.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace vue_spotify_app.Classes
{
    public class TrackFilter
    {
        public string Query { get; set; }

        public bool SearchName { get; set; }

        public bool SearchArtist { get; set; }

        public bool SearchAlbum { get; set; }

        public DateTime? DateRangeFrom { get; set; }

        public DateTime? DateRangeTo { get; set; }

        public SortType SortType { get; set; }

        public SortOrder SortOrder { get; set; }
    }
}

File: vue-spotify-app.client/src/classes/trackFilter.ts
import SortOrder from "@/enumClasses/sortOrder";
import SortType from "@/enumClasses/sortType";

export default class TrackFilter {

  query: string = "";
  searchName: boolean = true;
  searchArtist: boolean = true;
  searchAlbum: boolean = true;
  dateRangeFrom: string | null = null;
  dateRangeTo: string | null = null;
  sortType: number = SortType.Name;
  sortOrder: number = SortOrder.Ascending;
}
